userspace/libsinsp/event.cpp

```diff
@@ -2925,3 +2863,20 @@ std::optional<std::reference_wrapper<std::string>> sinsp_evt::get_enter_evt_para
 
 	return ret;
 }
+
+std::string sinsp_evt_param::invalid_len_error(size_t requested_length) const
```

Comment:
I have done some more experimentation. I know this is not pretty but by moving everything including the `throw` to a separate method we actually get the desired result (inlined `_as()` function with a call to the exception method, without that sometimes it would not be inlined). I also added the `cold` attribute although I'm not completely sure if it changes anything.